compiler/src/test/kotlin/batchtest.kt

```diff
@@ -97,7 +98,13 @@ class BatchTest {
         @ParameterizedTest
         @MethodSource("getNonCompilingPrograms")
         fun batchFail(filename: String, program: String) {
-            assertThrows<TerminatedCompilationException> {
+            try {
```

Comment:
I'm very confused as to why this test is written the way it's

Reply:
Okay. If `assertThrows` succeeds, the test passes. If it fails, it throws an AssertionFailedError - but wait - it might've been a parse-error and that's not encapsulated by TerminatedCompilationExceptions.

So the solution is to first try to see if a TerminatedCompilationException is thrown, if not, then compile again and assert for errors in either ErrorLogger or parser. This is what the comment specifies too. 

```kotlin
        fun batchFail(filename: String, program: String) {
            try {
                assertThrows<TerminatedCompilationException> {
                    compileTestProgram(program)
                }
            } catch (e: AssertionFailedError) {
                // If compilation does not throw a TerminatedCompilationException,
                // compile again and assert for errors in either ErrorLogger or parser
                val compileData = compileTestProgram(program)
                // Assert that errors are found
                assertTrue(
                    ErrorLogger.hasErrors() || compileData.parser.numberOfSyntaxErrors > 0,
                    "Non-compiling program compiled! Filename: $filename"
                )
            }
        }
```

Reply:
Either their is some poor design somewhere, or this is beyond me, either way I think I should let someone else review this bit

Reply:
Based on Cogitantium's comment: If `ErrorLogger.hasErrors()` is true, something is wrong, because there should have been thrown a `TerminatedCompilationException` earlier then, and the test should not pass. Otherwise, everything seems fine to me.

Reply:
Also, now it checks for failing to call `assertNoErrors()` at some point during compilation. Relevant commit: 4988b60. 
Please resolve if satisfied @jenrik.

Reply:
As I said earlier I'm letting somebody review this, please ask another team member for approval of this. I will dismiss my review when my other comments has been resolved.

Reply:
@NicEastvillage would you please approve this change?